### PR TITLE
Sample Plugin Source and Sink using Plugin support

### PR DESCRIPTION
**Pull request by gokulavasan**
Changes to pom files and modified Twitter Source, Stream Sink to use plugins framework. Tested manually.

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sinks/StreamSink.java [hunk 1]
```diff
@@ -76,12 +87,6 @@ public void configurePipeline(ETLStage stageConfig, PipelineConfigurer pipelineC
   @Override
   public void initialize(RealtimeContext context) throws Exception {
```
Comment (bdmogal) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sinks/StreamSink.java]: no need to override now, right?

File: cdap-app-templates/cdap-etl/cdap-etl-batch/pom.xml [hunk 2]
```diff
@@ -44,6 +44,7 @@
       <groupId>co.cask.cdap</groupId>
       <artifactId>cdap-etl-lib</artifactId>
       <version>${project.version}</version>
+      <scope>test</scope>
```
Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/pom.xml]: This is temporary, right? When we have the test framework supports adapter and plugin then we can remove this dependency entirely, right?

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sinks/StreamSink.java [hunk 3]
```diff
@@ -36,33 +38,42 @@
...
-                                        false));
+  private static final String NAME_DESC = "The name of the stream to output to. Must be a valid stream name. " +
+    "The stream will be created if it does not exist.";
+  private static final String DATA_FIELD_DESC = "Name of the field in the record that contains the data to be " +
+    "written to the specified stream. The data could be in binary format as a byte array or a ByteBuffer. " +
+    "It can also be a String. If unspecified, the 'data' key is used.";
+  private static final String HEADERS_FIELD_DESC = "Name of the field in the record that contains headers. " +
+    "Headers are presumed to be a map of string to string.";
+
+  private StreamConfig streamConfig;
```
Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sinks/StreamSink.java]: You can have this field `final` and have a constructor `StreamSink(StreamConfig config)` (you might need to turn `StreamConfig` class public for that.

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sources/TwitterSource.java [hunk 4]
```diff
@@ -68,19 +72,26 @@
...
-  @Override
-  public void configure(StageConfigurer configurer) {
-    configurer.setName(TwitterStreamSource.class.getSimpleName());
-    configurer.setDescription("Twitter Realtime Source");
-    configurer.addProperty(new Property(CONSUMER_KEY, "Consumer Key", true));
-    configurer.addProperty(new Property(CONSUMER_SECRET, "Consumer Secret", true));
-    configurer.addProperty(new Property(ACCESS_TOKEN, "Access Token", true));
-    configurer.addProperty(new Property(ACCESS_SECRET, "Access Token Secret", true));
+
+  private TwitterConfig twitterConfig;
```
Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sources/TwitterSource.java]: Same applies here

File: cdap-app-templates/cdap-etl/cdap-etl-batch/pom.xml (hunk 2 above)
Comment (gokulavasan) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/pom.xml]: Yes that is right.

**Comment by bdmogal** [comment]
:+1:

**Pull request merged**